core/reader.c

```diff
@@ -5,11 +5,20 @@ extern struct uwsgi_server uwsgi;
 int uwsgi_simple_wait_read_hook(int fd, int timeout) {
 	struct pollfd upoll;
 	timeout = timeout * 1000;
+        int ret = 0;
 
         upoll.fd = fd;
         upoll.events = POLLIN;
         upoll.revents = 0;
-        int ret = poll(&upoll, 1, timeout);
+        for(;;) {
```

Comment:
Doesn't it mean that we can stuck forever? If no data arrive, would we be able to kill the process with a signal?